### PR TITLE
[Intel MKL] Make Vanilla TF and MKL-based TF use common oneDNN build

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**

### CODE REVIEW

**Comment by penpornk** [comment]
@mahmoud-abuzaina Thank you for the PR!
https://github.com/tensorflow/tensorflow/pull/47543 might be reverted soon because of a nightly docker test build failure. I won't review this PR yet until we are sure the oneDNN v2.1 upgrade will stay.

**Comment by mahmoud-abuzaina** [comment]
@penpornk Thank you for the update. If you can share the oneDNN-related failure details, that would help us to look at it and fix it.

**Comment by gbaned** [comment]
@mahmoud-abuzaina Can you please resolve conflicts? Thanks!

**Comment by mahmoud-abuzaina** [comment]
@gbaned this PR depends on https://github.com/tensorflow/tensorflow/pull/47743. I will need to resolve the conflicts anyway once that is merged.

**Review by penpornk — APPROVED** [review]
Thank you for the PR!

**Review by penpornk — CHANGES_REQUESTED** [review]
The test failures are similar to those of PR #47743 (which needed headers in `include/**/*` for `dnnl_single_threaded`).

File: third_party/mkl_dnn/mkldnn_v1.BUILD [hunk 1]
```diff
@@ -121,51 +111,6 @@ cc_library(
...
-        "src/cpu/x64/xbyak",
-    ],
-    textual_hdrs = glob([
-        "include/**/*",
-        "src/common/*.hpp",
-        "src/cpu/*.hpp",
-        "src/cpu/**/*.hpp",
-        "src/cpu/x64/jit_utils/jitprofiling/*.h",
-        "src/cpu/x64/xbyak/*.h",
-    ]),
```
[the remark this answers is not in the document]
Comment (mahmoud-abuzaina) [reply third_party/mkl_dnn/mkldnn_v1.BUILD]: Sure. Done.

**Comment by mahmoud-abuzaina** [comment]
Thank you for pointing out the solution for the build error. I have made the changes.

**Review by penpornk — APPROVED** [review]
Thank you for the changes!

**Comment by gbaned** [comment]
@mahmoud-abuzaina  Can you please resolve conflicts? Thanks!

**Pull request merged**